docs/src/proposals/partitioned-inflationary-rewards-distribution.md

```diff
@@ -0,0 +1,110 @@
+---
+title: Partitioned Inflationary Rewards Distribution
+---
+
+## Problem
+
+With the increase of number of stake accounts, computing and redeeming the stake
+rewards at the start block of the epoch boundary becomes very expensive.
+Currently, with 550K stake accounts, the stake reward time has already taken
+more than 10 seconds. This prolonged computation slows down the network, and can
+cause large number of forks at the epoch boundary, which makes the matter even
+worse.
+
+## Proposed Solutions
+
+Instead of computing and reward stake accounts at epoch boundary, we will
+decouple reward computation and reward credit into two phases. At the start of
+the new epoch, a background thread is forked to compute the rewards. This is the
+first phase - reward computation. When reaching block height `N` after the
```

Comment:
What happens if we have a new block from an epoch crossing slot that has the parent of slot N-1?

Reply:
Good question. "At the start of the new epoch" is clear.
To clarify, for any block that cross the epoch boundaries, a new computation will be forked. 
In this case, we will have to fork another computation for N+2.
```
N-1 --- N -- N+1
      \
        N+2
```